core/store/models/vrfkey/types.go

```diff
@@ -0,0 +1,44 @@
+// Package vrfkey tracks the secret keys associated with VRF proofs. It
```

Comment:
Any reason why this isn't in the models package? Seems appropriate there, as these have gorm tags associated with them. Also, I found it confusing that the methods for the types are broken out across different files, that seems unconventional for Go. I think normally methods on and related to a type directly follow the type definition.

Reply:
Since we have two types of keys, now, it seemed sensible to put the new one in its own package. This way, its types are referred to as `vrfkey.T`, so they can't be confused with ethereum keys.

I put the types in their own file because I think it provides a good overview of the package and its functionality.

>  "Show me your flowcharts and conceal your tables, and I shall continue to be mystified. Show me your tables, and I won't usually need your flowcharts; they'll be obvious." -- Fred Brooks.

With jump-to-definition and instant live listing of struct fields, I don't find it particularly helpful to have struct definitions in the same file as its methods: I end up having to jump around the file anyway, so I might as well jump to a new file.

I'm not sure whether LSP allows you to list the methods on an object, though. I could make the public types interfaces, and put those in `types.go`, and then people could use jump-to-implementation to go to the method implementations.

Reply:
Unfortunately, those tools aren't available when reviewing PRs. 🙃

I'd prefer consistency and following language conventions to free form organization on the fly. I can't think of another Go project that organizes this way, and part of the goal of a standard for organizing things is to make it approachable for other people. As you've pointed out, discoverability is also lost when grouping things this way.

Reply:
OK, I'll move the types to their respective files, and add a `doc.go` to give an overview.